pocs/camera/sbigudrv.py

```diff
@@ -117,79 +74,188 @@ def retries(self, retries):
 
-        else:
-            # No serial number specified, just take the first unassigned handle
-            try:
-                index = self._handle_assigned.index(False)
-            except ValueError:
-                # All handles already assigned, must be trying to intialising more cameras than are
-                # connected.
-                self.logger.error('No connected SBIG cameras available!')
-                return (INVALID_HANDLE_VALUE, None)
+    def get_link_status(self):
+        lsr = GetLinkStatusResults()
+        with self._command_lock:
+            self._send_command('CC_GET_LINK_STATUS', params=lsr)
+        link_status = {'established': bool(lsr.linkEstablished),
+                       'base_address': int(lsr.baseAddress),
+                       'camera_type': camera_types[lsr.cameraType],
+                       'com_total': int(lsr.comTotal),
+                       'com_failed': int(lsr.comFailed)}
+        return link_status
+
+    def get_driver_handle(self):
+        ghr = GetDriverHandleResults()
+        with self._command_lock:
+            self._send_command('CC_GET_DRIVER_HANDLE', results=ghr)
+        return ghr.handle
 
-        handle = self._handles[index]
+    def set_handle(self, handle):
+        if not handle:
+            handle = INVALID_HANDLE_VALUE
+        set_handle_params = SetDriverHandleParams(handle)
+        self._send_command('CC_SET_DRIVER_HANDLE', params=set_handle_params)
 
-        self.logger.debug('Assigning handle {} to SBIG camera'.format(handle))
-        self._handle_assigned[index] = True
+    def get_ccd_info(self, handle):
+        """
+        Use Get CCD Info to gather all relevant info about CCD capabilities. Already
+        have camera type, 'name' and serial number, this gets the rest.
+        """
+        # 'CCD_INFO_IMAGING' will get firmware version, and a list of readout modes (binning)
+        # with corresponding image widths, heights, gains and also physical pixel width, height.
+        ccd_info_params0 = GetCCDInfoParams(ccd_info_request_codes['CCD_INFO_IMAGING'])
+        ccd_info_results0 = GetCCDInfoResults0()
 
-        # Get all the information from the camera
-        self.logger.debug('Obtaining SBIG camera info from {}'.format(handle))
-        ccd_info = self._get_ccd_info(handle)
+        # 'CCD_INFO_EXTENDED' will get bad column info, and whether the CCD has ABG or not.
+        ccd_info_params2 = GetCCDInfoParams(ccd_info_request_codes['CCD_INFO_EXTENDED'])
+        ccd_info_results2 = GetCCDInfoResults2()
 
-        # Serial number, name and type should match with those from Query USB Info obtained earlier
-        camera_serial = str(self._camera_info.usbInfo[index].serialNumber, encoding='ascii')
-        assert camera_serial == ccd_info['serial number'], \
-            self.logger.error('Serial number mismatch!')
+        # 'CCD_INFO_EXTENDED2_IMAGING' will info like full frame/frame transfer, interline or not,
```

Comment:
```suggestion
        # 'CCD_INFO_EXTENDED2_IMAGING' will return info like full frame/frame transfer, interline or not,
```
Something like that.

Reply:
I use 'get' for the others.